packages/lit-dev-content/site/tutorials/content/svg-templates/05.md

```diff
@@ -0,0 +1,170 @@
+including how to:
+
+- Style SVG with CSS
+- Use CSS Custom Properties to theme SVG
+
+### Learn
+
+CSS can apply presentation attributes to SVG elements similar to how 
+CSS applies attributes to HTML.
+
+However, only [SVG presentation attributes](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/Presentation)
+can be applied to SVG through CSS.
+
+In the following example, all `<rect>` elements will be as wide as
+their parent element while elements using the class `background` will
+have a black fill.
+
+
+```ts
+const helloSvgCss = css`
+	rect {
+		width: 100%;
+	}
+
+	.background {
+		fill: #000;
+	}
+`;
+```
+
+CSS classes can be applied to SVG Elements like in the example below.
+
+```ts
+const helloCssClasses = html`
+	<rect class="background"></rect>
+`;
+```
+
+SVG can also be styled with CSS Custom Properties. This enables artists
+and designers to theme an SVG document with the same styles as its
+HTML counterpart.
+
+```ts
+const helloCssCustomProperties = css`
+	.background {
+		fill: var(--background-color, #ff8800);
+	}
+`;
+```
+
+### Apply
+
+Add the following css template to `repeat-pattern.js`. It shouldn't affect
+`repeat-pattern` until the `static styles` property has been set.
+
+```ts
+import { css } from "lit";
+
+const svgCSS = css`
+	svg {
+		height: 100%;
+		width: 100%;
+	}
+
+	text {
+		fill: #fff;
+		dominant-baseline: hanging;
+		font-family: monospace;
+		font-size: 24px;
+	}
+
+	rect {
+		x: 0;
+		y: 0;
+		width: 100%;
+		height: 100%;
+	}
+`;
+```
+
+This demo requires a theme that includes:
+
+- `background-color`
+- `font-color`
+- `font-size`
+- `stroke-width`
+- `stroke-color`
+
+To accomplish this, create another CSS template with CSS custom properties
+representing the required theme.
+
+```ts
+const themeCSS = css`
+	.background {
+		fill: var(--background-color, #ff8800);
+	}
+
+	text {
+		fill: var(--font-color, #ffffff);
+		font-size: var(--font-size, 28px);
+		stroke-width: var(--stroke-width, 1.2);
+		stroke: var(--stroke-color, #0000dd);
+	}
+`;
+```
+
+Add the CSS templates to the `repeat-pattern` custom element.
+
+{% switchable-sample %}
+
+```ts
+@customElement('repeat-pattern')
+export class RepeatPattern extends LitElement {
+	static styles = [svgCSS, themeCSS];
+	...
+}
+```
+
+```js
+export class RepeatPattern extends LitElement {
+	static styles = [svgCSS, themeCSS];
+	...
+}
+customElements.define('repeat-pattern', RepeatPattern);
+```
+
+{% endswitchable-sample %}
+
+Next, add the `background` class to the `<rect>` representing the
+background element in our pattern.
+
+{% switchable-sample %}
+
+```ts
+@customElement('repeat-pattern')
+export class RepeatPattern extends LitElement {
+	static styles = [svgCSS, themeCSS];
+	...
+}
+```
+
+```js
+export class RepeatPattern extends LitElement {
+	static styles = [svgCSS, themeCSS];
+	...
+}
+customElements.define('repeat-pattern', RepeatPattern);
+```
+
+{% endswitchable-sample %}
```

Comment:
Watch out with this suggestion, it doesn't seem to be including the ending \`\`\`

```suggestion
```js
  render() {
    return html`
      ...
        <rect class="background" fill="#fff" width="100%" height="100%"></rect>
        <rect fill="url(#repeat-pattern)" width="100%" height="100%"></rect>
      </svg>
    `;
  }
```
```

Reply:
done!